qxf2_scheduler/routes.py

```diff
@@ -65,103 +69,177 @@ def index():
-                        each_interviewer['interviewer_end_time'] = [each_interviewer['interviewer_end_time'], interviewer_details['interviewer_end_time']]
-                     
-                    break
-             
-            if append_flag == True:                
-                interviewer_work_time_slots.append(interviewer_details)
-               
-    return render_template("list-interviewer.html", result=interviewer_work_time_slots)   
+            if interviewer_detail['interviewers_name'] in parsed_interviewer_details[0].values():
+                parsed_interviewer_details[0]['time'] = [
+                    parsed_interviewer_details[0]['time'], parsed_interviewer_detail['time']]
+
+    return parsed_interviewer_details
+
+
+@app.route("/<interviewer_id>/interviewer/")
+def read_interviewer_details(interviewer_id):
+    "Displays all the interviewer details"
+    # Fetching the Interviewer detail by joining the Interviewertimeslots tables and Interviewer tables
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+
+    parsed_interviewer_details = form_interviewer_details(interviewer_details)
+
+    return render_template("read-interviewers.html", result=parsed_interviewer_details)
+
+
+@app.route("/<interviewer_id>/interviewer/edit/", methods=['GET', 'POST'])
+def edit_interviewer(interviewer_id):
+    "Edit the interviewers"    
+    # This query fetch the interviewer details by joining the time slots table and interviewers table.
+    edit_interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).values(
+        Interviewers.interviewer_id, Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
     
+    parsed_interviewer_details = form_interviewer_details(edit_interviewer_details)
+
+    if request.method == "POST":
+        interviewer_name = request.form.get('name')
+        data = {'interviewer_name': interviewer_name}
+        print("ia interviewer is", interviewer_name, file=sys.stderr)
+        edit_interviewers = Interviewers.query.filter(Interviewers.interviewer_id == interviewer_id).update({'interviewer_name': request.form.get(
+            'name'), 'interviewer_email': request.form.get('email'), 'interviewer_designation': request.form.get('designation')})
+        db.session.commit()
+
+        # Adding the time slots in the interviewerstimeslots table
+        interviewer_time_slots = eval(request.form.get('timeObject'))
```

Comment:
I don't think you should use eval. Its not secure. Can you look for a more secure alternative?